Marlin/src/HAL/STM32/fastio.h

```diff
@@ -54,13 +54,14 @@ void FastIO_init(); // Must be called before using fast io macros
     if (V) FastIOPortMap[STM_PORT(digitalPin[IO])]->BSRR = _BV32(STM_PIN(digitalPin[IO])) ; \
     else   FastIOPortMap[STM_PORT(digitalPin[IO])]->BRR  = _BV32(STM_PIN(digitalPin[IO])) ; \
   }while(0)
+  #define _READ(IO)             bool(READ_BIT(FastIOPortMap[STM_PORT(digitalPin[IO])]->IDR, _BV32(STM_PIN(digitalPin[IO]))))
+  #define _TOGGLE(IO)           (FastIOPortMap[STM_PORT(digitalPin[IO])]->ODR ^= _BV32(STM_PIN(digitalPin[IO])))
 #else
-  #define _WRITE(IO, V) (FastIOPortMap[STM_PORT(digitalPin[IO])]->BSRR = _BV32(STM_PIN(digitalPin[IO]) + ((V) ? 0 : 16)))
+  #define _WRITE(IO, V)         (FastIOPortMap[STM_PORT(digitalPinToPinName(IO))]->BSRR = _BV32((digitalPinToPinName(IO) & 0xF) + ((V) ? 0 : 16)))
+  #define _READ(IO)             bool(READ_BIT(FastIOPortMap[STM_PORT(digitalPinToPinName(IO))]->IDR, _BV32(digitalPinToPinName(IO) & 0xF)))
```

Comment:
Why did you use:
`digitalPinToPinName(IO) & 0xF`

Instead of:
`STM_PIN(digitalPinToPinName(IO))`

Reply:
This was left over after copy-paste while searching for the problem. I am not yet very used to using macros and did not pay attention to it. The main thing for me was that it worked.